### PR TITLE
BUG: Compatible with openai 1.1

### PR DESCRIPTION
**Pull request by codingl2k1**
- Use `sse_starlette` handle fastapi stream response.

### CODE REVIEW

**Comment by aresnow1** [comment]
Are these changes compatible with older version of OpenAI? If not, we need to update our requirements.

File: xinference/api/restful_api.py [hunk 1]
```diff
@@ -811,15 +810,13 @@ async def stream_results():
                             prompt, system_prompt, chat_history, kwargs
                         )
                     async for item in iterator:
-                        yield f"data: {json.dumps(item)}\n"
+                        yield dict(data=json.dumps(item))
                 except Exception as ex:
```
Comment (UranusSeven) [inline xinference/api/restful_api.py:814]: Client disconnects need to be handled: https://github.com/sysid/sse-starlette/blob/159bbece1a58434d12d67934dc99fab556737247/examples/example.py#L62

Comment (codingl2k1) [reply xinference/api/restful_api.py:814]: I have handled all the exceptions including `asyncio.CancelledError`.

**Comment by codingl2k1** [comment]
> Are these changes compatible with older version of OpenAI? If not, we need to update our requirements.

It's compatible with old version of OpenAI.

**Comment by aresnow1** [comment]
The issue mentioned by comment is not introduced in this PR, merge this PR first and track this issue in #624

**Review by aresnow1 — APPROVED** [review]

**Pull request merged**